lib/routes/cool18/index.ts

```diff
@@ -3,73 +3,111 @@ import cache from '@/utils/cache';
 import got from '@/utils/got';
 import { load } from 'cheerio';
 import { parseDate } from '@/utils/parse-date';
+import type { Context } from 'hono';
+import ofetch from '@/utils/ofetch';
+
+type PageDataItem = {
+    tid: string;
+    username: string;
+    subject: string;
+    dateline: string;
+    type: string;
+};
 
 export const route: Route = {
-    path: '/:id?/:type?/:keyword?',
+    path: '/:id/:type?/:keyword?/:pageSize?',
+    url: 'cool18.com',
+    example: 'cool18.com/bbs4',
+    parameters: {
+        id: 'the name of the bbs',
+        type: 'the type of the post. Can be `home`, `gold` or `threadsearch`. Default: `home`',
+        keyword: 'the keyword to search.',
+        pageSize: 'the number of posts to fetch. If the type is not in search, you can type any words. Default: 10',
+    },
+    categories: ['bbs'],
     radar: [
         {
-            source: ['cool18.com/'],
+            source: ['cool18.com/:id/', 'www.cool18.com/:id/index.php?action=search&bbsdr=:id&act=:type&app=forum&keywords=:keyword&submit=%E6%9F%A5%E8%AF%A2', 'www.cool18.com/:id/index.php?app=forum&act=:type'],
+            target: '/:id/:type?/:keyword?',
         },
     ],
-    name: 'Unknown',
-    maintainers: ['nczitzk'],
+    name: '留园网',
+    maintainers: ['nczitzk', 'Gabrlie'],
     handler,
-    url: 'cool18.com/',
 };
 
-async function handler(ctx) {
-    const id = ctx.req.param('id') ?? 'bbs4';
-    const type = ctx.req.param('type') ?? '';
-    const keyword = ctx.req.param('keyword') ?? '';
+async function handler(ctx: Context) {
+    const { id, type = 'home', keyword, pageSize = '10' } = ctx.req.param();
```

Comment:
No need to add additional path parameter `pageSize` when everything can be done using [common search parameter](https://docs.rsshub.app/guide/parameters#limit-entries) limit.https://github.com/DIYgod/RSSHub/blob/0510d8391e08b2260787ca7955c88fd18ea4d478/lib/routes/cool18/index.ts#L37